overlays/nodejs/sqs/sqsMixins.ts

```diff
@@ -0,0 +1,120 @@
+}
+
+export interface QueueRecord {
+    messageId: string;
+    receiptHandle: string;
+    body: string;
+    attributes: {
+        ApproximateReceiveCount: string;
+        SentTimestamp: string;
+        SenderId: string;
+        ApproximateFirstReceiveTimestamp: string;
+    };
+    messageAttributes: Record<string, any>;
+    md5OfBody: string;
+    eventSource: string;
+    eventSourceARN: string;
+    awsRegion: string;
+}
+
+export type QueueEventHandler = lambda.EventHandler<QueueEvent, void>;
+
+/**
+ * Arguments to control the sqs subscription.
+ */
+export type QueueEventSubscriptionArgs = {
+    /**
+     * The largest number of records that AWS Lambda will retrieve. The maximum batch size supported
+     * by Amazon Simple Queue Service is up to 10 queue messages per batch. The default setting is
+     * 10.
+     *
+     * See https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html for more details.
+     */
+    batchSize?: number;
+ };
+
+export class QueueEventSubscription extends lambda.EventSubscription {
+    public readonly queue: queue.Queue;
+
+    /**
+     * The underlying sns object created for the subscription.
+     */
+    public readonly eventSourceMapping: lambda.EventSourceMapping;
+
+    public constructor(
+        name: string, queue: queue.Queue, handler: QueueEventHandler,
+        args: QueueEventSubscriptionArgs, opts?: pulumi.ResourceOptions) {
+
+        super("aws:sqs:QueueEventSubscription", name, { }, opts);
+
+        args = args || {};
+
+        const parentOpts = { parent: this };
+        this.func = createFunctionFromEventHandler(name, handler, parentOpts);
+
+        this.permission = new lambda.Permission(name, {
+            action: "lambda:*",
+            function: this.func,
+            principal: "sqs.amazonaws.com",
+            sourceArn: queue.arn,
+        }, parentOpts);
+
+        this.eventSourceMapping = new lambda.EventSourceMapping(name, {
+            batchSize: args.batchSize,
+            enabled: true,
+            eventSourceArn: queue.arn,
+            functionName: this.func.name,
+        }, parentOpts);
+    }
+}
+
+function createFunctionFromEventHandler(
+    name: string, handler: QueueEventHandler, opts?: pulumi.ResourceOptions): lambda.Function {
+
+    if (handler instanceof Function) {
+        return lambda.createFunction(name, {
+            entryPoint: handler,
+            policies: [iam.AWSLambdaFullAccess, iam.AmazonSQSFullAccess]
+        } , opts);
+    }
+    else {
+        return handler;
+    }
+}
+
+declare module "./queue" {
+    interface Queue {
+        /**
+         * Creates a new subscription to events fired from this Queue to the handler provided, along
+         * with options to control the behavior of the subscription.
+         */
+        onEvent(
+            name: string, handler: QueueEventHandler,
+            args?: QueueEventSubscriptionArgs, opts?: pulumi.ResourceOptions): QueueEventSubscription;
```

Comment:
bikeshedding. how does .onEvent sound to people here?   we could say .onNewMessages, but i don't think there's high value in changing hte name.